earthly: init at 0.5.11

zoedsoupe opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Earthly can help to generate repeatable and understandable builds on CI or locally.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Commit:
earthly: init at 0.5.11

Update pkgs/top-level/all-packages.nix; Added earthly

Do check tests

created personal handle in maintainers/maintainer-list.nix

preppended "debugger" and "shellrepeater" binaries with earthly-* to avoid collisions

fixed earthly package maintainer name

SuperSandro2000 commented on pkgs/development/tools/earthly/default.nix:
```diff
@@ -0,0 +1,25 @@
+    sha256 = "1d9p2f79f2k7nnka9qja3dlqvvl240l09frkb17ff2f5kyi1qabv";
+  };
+
+  doCheck = false;
```
Why did you disable tests?

Mdsp9070 replied:
Yeah, you're right, no reason at all.

NobbZ commented:
Can this get in before 21.05 branch off?

I really need `earthly` for a customer project, and I managed to defer the rewrite of related stuff until next month with "earthly is not officially supported on my linux distro currently but should be with the next release in about a month".

I'd be happy if the PR gets at least merged into master/unstable by then, but 21.05 would be much more appreciated.

bryanasdev000 reviewed (approved):
LGTM, I did some quick tests, it looks like I have a new tool in my toolbox.
The only thing I think would help is a post-installation phase to rename `debugger` and` shellrepeater` to something like `earthly- *` to not collide with other random binaries.

bryanasdev000 commented:
@Mdsp9070 you need to add yourself to `maintainers/maintainer-list.nix` to fix errors of offborg.

Also, seems that `pkgs/top-level/all-packages.nix` have a Trailing whitespace at 2406.

bryanasdev000 commented:
> Can this get in before 21.05 branch off?
> 
> I really need `earthly` for a customer project, and I managed to defer the rewrite of related stuff until next month with "earthly is not officially supported on my linux distro currently but should be with the next release in about a month".
> 
> I'd be happy if the PR gets at least merged into master/unstable by then, but 21.05 would be much more appreciated.

I don't think it's a problem, when @Mdsp9070  fixes the question of maintainers we can ping here https://discourse.nixos.org/t/prs-already-reviewed/2617/375 or on discord to speed up the merge.

We have until May 21st according to https://discourse.nixos.org/t/21-05-release-schedule/12528 to 21.05 release.

Mdsp9070 commented:
> @Mdsp9070 you need to add yourself to `maintainers/maintainer-list.nix` to fix errors of offborg.
> 
> Also, seems that `pkgs/top-level/all-packages.nix` have a Trailing whitespace at 2406.

I added myself to `maintainers/maintainer-list.nix`; however I didn't find this trainling white space... I executed with nixpkgs-fmt but it changed so many lines so I gave up.

Mdsp9070 commented:
> LGTM, I did some quick tests, it looks like I have a new tool in my toolbox.
> The only thing I think would help is a post-installation phase to rename `debugger` and` shellrepeater` to something like `earthly- *` to not collide with other random binaries.

How I could make this? I'm pretty new to Nix derivations.

I tried:
```nix
postInstall = ''
  mv $src/cmd/debugger $src/cmd/earthly-debugger
  mv $src/cmd/shellrepeater $src/cmd/earthly-shellrepeater
'';
```

But this obviously fails with `Permisson denied`.

the `$out` directory isn't only for the binary output?

bryanasdev000 reviewed (approved):
LGTM, you will probably need to squash before merge.

bryanasdev000 commented on pkgs/development/tools/earthly/default.nix:
```diff
@@ -0,0 +1,28 @@
+    homepage = "https://earthly.dev/";
+    changelog = "https://github.com/earthly/earthly/releases/tag/v${version}";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ Mdsp9070 ];
```
This need to be `mdsp` the same as in `maintainer-list.nix`.

bryanasdev000 reviewed (commented):
Maintainer name;

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -11162,6 +11162,12 @@
     githubId = 26458780;
     name = "Matthew Croughan";
   };
+  mdsp = {
```
Please put this into a separate commit with the message ``maintainers: add mdsp`` and sort it alphabetically.

Commit:
maintainers: add mdsp

removed non alphabetically sorted mdsp maintainer

remove trailing whitespace in pkgs/top-level/all-packages.nix:2377

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -2374,6 +2374,8 @@ in
 
   dyndnsc = callPackage ../applications/networking/dyndns/dyndnsc { };
 
+  earthly = callPackage ../development/tools/earthly { }; 
```
```suggestion
  earthly = callPackage ../development/tools/earthly { };
```
